src/app/identity.service.ts

```diff
@@ -67,6 +72,24 @@ export class IdentityService {
     this.cast('login', payload);
   }
 
+  derive(payload: {
+    derivedUserInfo: DerivedUserInfo,
+  }): void {
+    if (this.globalVars.callback) {
+      // If callback is passed, we redirect to it with payload as URL parameters.
+      let paramString = '?';
+      for (const key in payload.derivedUserInfo) {
+        if (payload.derivedUserInfo.hasOwnProperty(key)) {
+          paramString += key + '=' + (payload.derivedUserInfo as any)[key].toString() + '&';
+        }
+      }
+      paramString = paramString.slice(0, -1);
```

Comment:
nit: This is a bit clever, which is generally bad but acceptable in this case... It just so happens that there will either be a ? or an & at the end. I would add a comment explaining tihs cleverness. It takes some brianpower to figure out why this won't break in certain caises. When that's the case, always add a comment to explain the situation. This helps us avoid breaking it in the future, or at least helps us understand why it broke if we're coming back to it.